treeherder/client/thclient/client.py

```diff
@@ -701,6 +705,38 @@ def _post_json(self, project, endpoint, oauth_key, oauth_secret, jsondata,
                              timeout=timeout)
         resp.raise_for_status()
 
+    def _get_json(self, project, endpoint, timeout=None):
+
+        if timeout is None:
+            timeout = self.timeout
+        response = requests.get(self._get_uri(project, endpoint), timeout=timeout)
+        response.raise_for_status()
+        return response.json()
+
+    def query_resultset_api(self, project, **filter_params):
+        """
+        Query the /resultset/ api of treeherder
+
+        :param project: project (repository name) to query data for
+        :param filter_params: keyword arguments to filter results
+        """
+
```

Comment:
Hmm, I like the general idea here. The terminology/naming seems a bit off -- the whole point of a client is to abstract away some of the details of interacting with the API, by including that nomenclature in the name we're defeating that.

Let's do something like:

    def get_resultsets(self, project, **params):
       """
       Gets resultsets from project, filtered by parameters

       By default this method will just return the latest 10 result sets (if they exist)

       :param project: project (repository name) to query data for
       :param filter_params: keyword arguments to filter results
        """